open_spiel/games/bridge/bridge.cc

```diff
@@ -357,13 +350,42 @@ void BridgeState::WriteObservationTensor(Player player,
         ptr[relative_player * kNumCards + card] = 1;
       }
     }
+
     ptr += kNumPlayers * kNumCards;
 
+    // Previous tricks
+    for (int j = current_trick - 1; j >= std::max(0, current_trick - num_tricks_ + 1); --j) {
+      int leader = tricks_[j].Leader();
+      for (int i = 0; i < kNumPlayers; ++i) {
+        int card = history_[this_trick_start - kNumPlayers * (current_trick - j) + i].action;
+        int relative_player = (i + leader + kNumPlayers - player) % kNumPlayers;
+        ptr[relative_player * kNumCards + card] = 1;
+      }
+      ptr += kNumPlayers * kNumCards;
+    }
+
+    // Move pointer for future tricks to have a fixed size tensor
+    if (num_tricks_ > current_trick + 1) {
+      ptr += kNumPlayers * kNumCards * (num_tricks_ - current_trick - 1);
+    }
+
     // Number of tricks taken by each side.
     ptr[num_declarer_tricks_] = 1;
     ptr += kNumTricks;
     ptr[num_cards_played_ / 4 - num_declarer_tricks_] = 1;
     ptr += kNumTricks;
+
+    int kPlayTensorSize =
+        kNumBidLevels                           // What the contract is
+        + kNumDenominations                     // What trumps are
+        + kNumOtherCalls                        // Undoubled / doubled / redoubled
+        + kNumPlayers                           // Who declarer is
+        + kNumVulnerabilities                   // Vulnerability of the declaring side
+        + kNumCards                             // Our remaining cards
+        + kNumCards                             // Dummy's remaining cards
+        + num_tricks_ * kNumPlayers * kNumCards // Number of played tricks
+        + kNumTricks                            // Number of tricks we have won
+        + kNumTricks;                           // Number of tricks they have won
```

Comment:
Might be a good idea to make this a function, since you do the same calculation in `GetObservationTensorSize`.